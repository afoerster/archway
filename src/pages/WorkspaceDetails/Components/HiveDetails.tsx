import { Card, Icon, List } from 'antd';
import * as React from 'react';
import { HiveTable, NamespaceInfo } from '../../../types/Workspace';
import Label from './Label';
import { HueService } from '../../../types/Cluster';
import Colors from '../../../components/Colors';

interface Props {
    hue?: HueService;
    namespace: string;
    info?: NamespaceInfo[];
}

const renderTable = (hiveTable: HiveTable) => (
  <div style={{ margin: 10, textAlign: 'center' }}>{hiveTable.name}</div>
);

const HiveDetails = ({ hue, namespace, info }: Props) => (
  <Card
    actions={[
      <a href={hue && `http://${hue.load_balancer[0].host}:${hue.load_balancer[0].port}/hue/metastore/tables/`}>
        See in Hue
      </a>,
    ]}>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
      <Label style={{ lineHeight: '20px' }}>
        <Icon
          theme="twoTone"
          twoToneColor={Colors.Green.string()}
          type="database"
          style={{ paddingRight: 5, fontSize: 20 }} />Hive
        </Label>
      <Label style={{ lineHeight: '18px', fontSize: 10 }}>
        {namespace}
      </Label>
    </div>
    <List
      dataSource={info && info[0] && info[0].tables}
      renderItem={renderTable}
      locale={{ emptyText: 'No tables yet' }} />
  </Card>
);

export default HiveDetails;
