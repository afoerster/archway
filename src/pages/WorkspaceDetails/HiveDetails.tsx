import * as React from 'react';
import { HiveTable } from '../../types/Workspace';
import { List } from 'antd';

interface Props {
  namespace: string;
  tables?: HiveTable[];
}

const renderTable = (hiveTable: HiveTable) => (
  <div>{hiveTable.name}</div>
);

const HiveDetails = ({ namespace, tables }: Props) => (
  <List
    dataSource={tables}
    renderItem={renderTable} />
);

export default HiveDetails;
