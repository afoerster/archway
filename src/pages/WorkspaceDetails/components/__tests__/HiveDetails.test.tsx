import * as React from 'react';
import { shallow } from 'enzyme';

import HiveDetails from '../HiveDetails';

describe('HiveDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HiveDetails namespace="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
