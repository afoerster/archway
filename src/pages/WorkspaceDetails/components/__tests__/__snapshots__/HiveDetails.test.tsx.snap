// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiveDetails renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HiveDetails
    namespace=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <a
          target="_blank"
        >
          See in Hue
        </a>,
      ],
      "children": Array [
        <CardHeader
          heading="Hive Tables"
          icon="database"
          subheading=""
        />,
        <Row
          gutter={12}
          justify="center"
          style={
            Object {
              "marginTop": 18,
            }
          }
          type="flex"
        >
          <div
            style={
              Object {
                "color": "rgba(0, 0, 0, .65)",
              }
            }
          >
            No tables yet.
          </div>
        </Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "heading": "Hive Tables",
          "icon": "database",
          "subheading": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            undefined,
            <div
              style={
                Object {
                  "color": "rgba(0, 0, 0, .65)",
                }
              }
            >
              No tables yet.
            </div>,
          ],
          "gutter": 12,
          "justify": "center",
          "style": Object {
            "marginTop": 18,
          },
          "type": "flex",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No tables yet.",
              "style": Object {
                "color": "rgba(0, 0, 0, .65)",
              },
            },
            "ref": null,
            "rendered": "No tables yet.",
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <a
            target="_blank"
          >
            See in Hue
          </a>,
        ],
        "children": Array [
          <CardHeader
            heading="Hive Tables"
            icon="database"
            subheading=""
          />,
          <Row
            gutter={12}
            justify="center"
            style={
              Object {
                "marginTop": 18,
              }
            }
            type="flex"
          >
            <div
              style={
                Object {
                  "color": "rgba(0, 0, 0, .65)",
                }
              }
            >
              No tables yet.
            </div>
          </Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "heading": "Hive Tables",
            "icon": "database",
            "subheading": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              undefined,
              <div
                style={
                  Object {
                    "color": "rgba(0, 0, 0, .65)",
                  }
                }
              >
                No tables yet.
              </div>,
            ],
            "gutter": 12,
            "justify": "center",
            "style": Object {
              "marginTop": 18,
            },
            "type": "flex",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No tables yet.",
                "style": Object {
                  "color": "rgba(0, 0, 0, .65)",
                },
              },
              "ref": null,
              "rendered": "No tables yet.",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
